test/FunctionalTests/Microsoft.OData.Core.Tests/IntegrationTests/Evaluation/AutoComputePayloadMetadataInJsonIntegrationTests.cs

```diff
@@ -753,6 +752,168 @@ public void WritingDynamicComplexPropertyWithModelSpecifiedInFullMetadataMode_40
+            InMemoryMessage message = new InMemoryMessage();
+            message.SetHeader("Content-Type", "application/json;odata.metadata=minimal");
+            message.Stream = new MemoryStream(Encoding.UTF8.GetBytes(payload));
+
+            ODataResource topLevelResource = null;
+            IList<ODataResource> nestedResources = new List<ODataResource>();
+            int deep = 0;
+            using (var messageReader = new ODataMessageReader((IODataResponseMessage)message, null, model))
+            {
+                var reader = messageReader.CreateODataResourceReader(entitySet, entityType);
+                while (reader.Read())
+                {
+                    switch (reader.State)
+                    {
+                        case ODataReaderState.ResourceStart:
+                            deep++;
+                            break;
+                        case ODataReaderState.ResourceEnd:
+                            if (deep == 2)
+                            {
+                                nestedResources.Add((ODataResource)reader.Item);
+                            }
+                            else
+                            {
+                                topLevelResource = (ODataResource)reader.Item;
+                            }
+                            deep--;
+                            break;
+                    }
+                }
+            }
+
+            Assert.NotNull(topLevelResource);
+            Assert.Equal(new Uri("http://svc/Customers(9)"), topLevelResource.Id);
+
+            Assert.Equal(2, nestedResources.Count);
+            Assert.Equal(new[] { "NS.Address", "NS.Customer" }, nestedResources.Select(n => n.TypeName));
+
+            // #1
+            ODataResource nestedResource = nestedResources.First(c => c.TypeName == "NS.Address");
+            Assert.Equal(2, nestedResource.Properties.Count());
+            Assert.Equal("Redmond", nestedResource.Properties.First(p => p.Name == "City").Value);
+
+            // #2
+            nestedResource = nestedResources.First(c => c.TypeName == "NS.Customer");
+            Assert.Equal(1, nestedResource.Properties.Count());
+            Assert.Equal(42, nestedResource.Properties.First(p => p.Name == "Id").Value);
+        }
+
+        private static IEdmModel GetModelWithAbstractType(out EdmEntitySet entitySet)
+        {
+            var model = new EdmModel();
+            var complexType = new EdmComplexType("NS", "Address");
+            complexType.AddStructuralProperty("City", EdmPrimitiveTypeKind.String);
+            complexType.AddStructuralProperty("ZipCode", EdmPrimitiveTypeKind.Int32);
+            var customer = new EdmEntityType("NS", "Customer");
+            customer.AddKeys(customer.AddStructuralProperty("Id", EdmPrimitiveTypeKind.Int32));
+            // <Property Name="ComplexData" Type="Edm.ComplexType" />
+            customer.AddStructuralProperty("ComplexData", EdmCoreModel.Instance.GetComplexType(true));

```

Comment:
>ComplexData [](start = 44, length = 11)

maybe "ComplexProperty" and "EntityProperty"? just a personal preference; if you prefer the "Data" suffix that's okay with me, I just think the "Property" suffix makes this code a bit more readable.

Reply:
#Resolved.